include/mxnet/c_api.h

```diff
@@ -1833,10 +1832,12 @@ MXNET_DLL int MXExecutorGetOptimizedSymbol(ExecutorHandle handle,
 
 /*!
  * \brief set a call back to notify the completion of operation
+ * \param monitor_all If true, monitor both input and output, otherwise monitor output only.
  */
 MXNET_DLL int MXExecutorSetMonitorCallback(ExecutorHandle handle,
                                            ExecutorMonitorCallback callback,
-                                           void* callback_handle);
+                                           void* callback_handle,
+                                           bool monitor_all);
```

Comment:
default value is added in python interface. I'm not sure if we encourage adding default value for c api. There's no precedent. @KellenSunderland @TaoLv Do you agree to add default value for c_api?

Reply:
I think it might be useful for people directly using the c_api to have a default value here.  Generally as I mentioned above I believe MXNet only cares about user APIs.  However given this API has been around for a long time and users often use the c_api I'd rather be safe than sorry.

Reply:
According to https://stackoverflow.com/questions/1472138/c-default-arguments, we shouldn't add default value for this file as it defined C api instead of C++. So I removed the default value.